src/coredump/coredump.c

```diff
@@ -1212,8 +1218,25 @@ static int process_kernel(int argc, char* argv[]) {
 
         if (context.is_journald || context.is_pid1)
                 r = submit_coredump(&context, iovw, STDIN_FILENO);
-        else
+        else {
+                _cleanup_free_ char *process_name;
+                process_name = NULL;
+                process_name = filename_escape(context.meta[META_COMM]);
+                if (!process_name)
+                        log_error_errno(r, "Failed to escape process name. Ignoring coredump ratelimit: %m");
+                else {
+                        /* per process coredump ratelimit */
+                        r = coredump_ratelimit(process_name, arg_ratelimitinterval, arg_ratelimitburst);
+                        if (r < 0)
+                                log_error_errno(r, "Failed to process coredump ratelimit: %m");
+
+                        if (r > 0) {
+                                log_warning("Core dump of %s is ratelimited", process_name);
+                                goto finish;
```

Comment:
should propagate 0.

Reply:
@Sribharath27 this is still unsolved - please set ```r = 0;``` before the goto

Reply:
Thank you for the comment. Addressed the comment. Please check once